### PR TITLE
heater settings for other types of sensors

### PR DESCRIPTION
**Pull request by rusefillc**

### CODE REVIEW

**Comment by rusefillc** [comment]
@dron0gus FYI :)

File: firmware/boards/port.h [hunk 1]
```diff
@@ -18,7 +18,8 @@ struct AnalogResult
 
 AnalogResult AnalogSample();
 
-enum class SensorType : uint8_t {
+//enum class SensorType : uint8_t {
+enum SensorType {
```
Comment (mck1117) [inline firmware/boards/port.h:22]: why change?

File: firmware/heater_control.cpp [hunk 2]
```diff
@@ -33,6 +62,7 @@ static const PWMConfig heaterPwmConfig = {
 
 static constexpr int preheatTimeCounter = HEATER_PREHEAT_TIME / HEATER_CONTROL_PERIOD;
 static constexpr int batteryStabTimeCounter = HEATER_BATTERY_STAB_TIME / HEATER_CONTROL_PERIOD;
+static const struct sensorHeaterParams *heater;
```
Comment (mck1117) [inline firmware/heater_control.cpp:65]: nit: no `struct`
nit: name that describes that this is params

File: firmware/boards/port.h (hunk 1 above)
Comment (dron0gus) [reply firmware/boards/port.h:22]: Couse Im not a C++ guru and don't know how to workaround:
```
static const struct sensorHeaterParams heaterParams[] = {
    [SensorType::LSU49] = {
        //LSU4.9
        .closedLoopThresholdESR = LSU49_HEATER_CLOSED_LOOP_THRESHOLD_ESR,
        .targetESR = LSU49_HEATER_TARGET_ESR,
        .overheatESR = LSU49_HEATER_OVERHEAT_ESR,
        .underheatESR = LSU49_HEATER_UNDERHEAT_ESR,
    },
    [SensorType::LSU42] = {
        //LSU4.2
        .closedLoopThresholdESR = LSU42_HEATER_CLOSED_LOOP_THRESHOLD_ESR,
        .targetESR = LSU42_HEATER_TARGET_ESR,
        .overheatESR = LSU42_HEATER_OVERHEAT_ESR,
        .underheatESR = LSU42_HEATER_UNDERHEAT_ESR,
    },
    [SensorType::LSUADV] = {
        //LSU_ADV
        //TODO
    }
};
```
Causing:
```
heater_control.cpp:22:18: error: C99 designator 'SensorType::LSU49' is not an integral constant-expression
   22 |     [SensorType::LSU49] = {
      |      ~~~~~~~~~~~~^~~~~
heater_control.cpp:29:18: error: C99 designator 'SensorType::LSU42' is not an integral constant-expression
   29 |     [SensorType::LSU42] = {
      |      ~~~~~~~~~~~~^~~~~
heater_control.cpp:36:18: error: C99 designator 'SensorType::LSUADV' is not an integral constant-expression
   36 |     [SensorType::LSUADV] = {
      |      ~~~~~~~~~~~~^~~~~~
```

Comment (dron0gus) [reply firmware/boards/port.h:22]: Also `LSUADV` looks clearer than `SensorType::LSUADV`

Comment (rusefillc) [reply firmware/boards/port.h:22]: @mck1117 how about https://github.com/mck1117/wideband/pull/183?

File: firmware/boards/f1_dual/io/io_pins.h [hunk 3]
```diff
@@ -22,13 +22,28 @@
...
+#define NERNST_49_ESR_DRIVER_PIN	11
+
+#define NERNST_49_BIAS_PORT			GPIOB
+#define NERNST_49_BIAS_PIN			2
 
 // LSU ADV - 47K
 #define NERNST_ADV_ESR_DRIVER_PORT	GPIOB
 #define NERNST_ADV_ESR_DRIVER_PIN	10
 
+/* This is temporaly solution, waiting for sensor type saved in settings */
```
Comment (mck1117) [inline firmware/boards/f1_dual/io/io_pins.h:35]: we have sensor type stored in settings, do we need temporary solution?

Comment (dron0gus) [reply firmware/boards/f1_dual/io/io_pins.h:35]: We do not have support from ini/TS side yet.

Comment (mck1117) [reply firmware/boards/f1_dual/io/io_pins.h:35]: Yes we do: https://github.com/mck1117/wideband/blob/62520aee781483fb0a9c4792f0e972c60956d172/firmware/ini/rusefi_wb_f1.ini#L193

**Comment by mck1117** [comment]
does #207 supercede this?

**Pull request closed without merging**